syntax/parser.go

```diff
@@ -921,6 +956,10 @@ func (p *parser) scanGroupOpen() (*regexNode, error) {
 					if p.charsRight() > 0 && !(p.rightChar(0) == close || p.rightChar(0) == '-') {
 						return nil, p.getErr(ErrInvalidGroupName)
 					}
+
+					if capnum != -1 && p.useMaintainCaptureOrder() {
```

Comment:
a comment here would be nice, why do consume the Autocap?  I assume it's because we've just successfully setup a named capture group so we need to increment our cap numbers to maintain the order in this mode